src/editor.c

```diff
@@ -1843,28 +1843,9 @@ static gint find_start_bracket(ScintillaObject *sci, gint pos)
 }
 
 
-static gchar *find_calltip(const gchar *word, GeanyFiletype *ft)
+static GPtrArray *update_tags_for_constructor_calltip(GPtrArray *tags, GeanyFiletype *ft,
+													  TMTag *tag, const gchar *constructor_method)
```

Comment:
I think it's a bit confusing for the reader of the code to see what this function does - it does a bit too many things. I think it would be better to call it something like `get_constructor_tags()` without passing the `tags` argument to it so this function would return `GPtrArray *` when constructors are found or NULL otherwise. Then, later, when this function is called, the code would look like
```C
GPtrArray *constructor_tags = get_constructor_tags(ft, tag, constructor_method);
if (constructor_tags)
{
    g_ptr_array_free(tags, TRUE);
    tags = constructor_tags;
}
```

Reply:
Good point, done.